src/cpp/fastdds/core/policy/QosPoliciesSerializer.hpp

```diff
@@ -1296,8 +1330,11 @@ inline bool QosPoliciesSerializer<RTPSEndpointQos>::add_to_cdr_message(
         const RTPSEndpointQos& qos_policy,
         rtps::CDRMessage_t* cdr_message)
 {
-    bool valid = true;
+    // Add common to cdr_message
+    bool valid = rtps::CDRMessage::addUInt16(cdr_message, PID_RTPS_ENDPOINT);
+    valid &= rtps::CDRMessage::addUInt16(cdr_message, static_cast<uint16_t>(cdr_serialized_size(qos_policy) - 4));
```

Comment:
For the next major we could also add a `TODO` of making all eprosima extended Qos to inherit from `QosPolicy`. Currently only some of them are doing so. What do you think ?